src/hashes/sha3_test.c

```diff
@@ -388,161 +388,256 @@ int sha3_shake_test(void)
 #endif
 }
 
+#if defined LTC_TURBO_SHAKE || defined LTC_KT
+static LTC_INLINE int s_turbo_shake_generate_ptn(unsigned char* buffer, long offset, long amount)
+{
+   long i;
+
+   LTC_ARGCHK(buffer != NULL || amount == 0);
+   LTC_ARGCHK(offset >= 0);
+   LTC_ARGCHK(amount >= 0);
+
+   for(i = 0; i != amount; ++i)
+   {
+      buffer[i] = ((unsigned char)(((offset + i) % 0xfb) & 0xff));
+   }
+   return CRYPT_OK;
+}
+static LTC_INLINE unsigned char s_turbo_shake_hex_to_nibble(char hex)
+{
+   LTC_ARGCHK((hex >= '0' && hex <= '9') || (hex >= 'a' && hex <= 'f') || (hex >= 'A' && hex <= 'F'));
+
+   return (hex >= '0' && hex <= '9') ? (hex - '0') : ((hex >= 'a' && hex <= 'f') ? (10 + hex - 'a') : (10 + hex - 'A'));
+}
+static LTC_INLINE int s_turbo_shake_hex_to_bin(char const* hex, unsigned char* bin, long amount)
+{
+   long i;
+
+   LTC_ARGCHK(hex != NULL || amount == 0);
+   LTC_ARGCHK(bin != NULL || amount == 0);
+   LTC_ARGCHK(amount >= 0);
+
+   for(i = 0; i != amount; ++i)
+   {
+      bin[i] =
+         (s_turbo_shake_hex_to_nibble(hex[i * 2 + 0]) << 4) |
+         (s_turbo_shake_hex_to_nibble(hex[i * 2 + 1]) << 0);
+   }
+   return CRYPT_OK;
+}
+#endif
+
+#ifdef LTC_TURBO_SHAKE
+static LTC_INLINE int s_turbo_shake_test_one(int bits_count, long input_bytes_count, long skip_digest_bytes, long digest_bytes_count, int counter, char const *expected_digest_hex)
```

Comment:
I very much like the idea where this is going to! 👍 💯 

How about:


```suggestion
typedef struct turbo_shake_test {
   int bits_count;
   unsigned long input_bytes_count, skip_digest_bytes, digest_bytes_count;
   const char *expected_digest_hex;
} turbo_shake_test;


static LTC_INLINE int s_turbo_shake_test_one(const turbo_shake_test *testcase, int counter)
```

Reply:
Yes, fixed.